test/common/network/dns_impl_test.cc

```diff
@@ -1,3 +1,5 @@
+#include <ares.h>
```

Comment:
Still not a big fan of this; it's inconsistent with the way we treat all other external dependencies, and the user might easily believe its coming from the system vs. the gRPC external dependency. Is there no way we can improve this (@jmillikin-stripe)? If not, we should put comments at each of these include sites to explain what's going on and the nature of this exception..

Reply:
Do you think it'd make sense to use `#include <>` style for all external dependencies? The presence of "local"-style includes of external deps is fairly strange for C++ code, and IMO is an unwanted abstraction leak from the build system into the code being built.